src/components/dateTimePicker/index.tsx

```diff
@@ -173,17 +172,29 @@ function DateTimePicker(props: DateTimePickerPropsInternal) {
     if (value) {
       switch (mode) {
         case MODES.DATE:
-          return dateFormatter
-            ? dateFormatter(value)
-            : dateFormat
-              ? moment(value).format(dateFormat)
-              : value.toLocaleDateString();
+          if (dateFormatter) {
+            return dateFormatter(value);
+          }
+          if (!dateFormat) {
+            return value.toLocaleDateString();
+          }
+          if (!moment) {
+            console.error(`RNUILib DateTimePicker component with dateFormat requires installing "moment" dependency`);
```

Comment:
I think adding this and the other `error` to a `useEffect` (either the existing one or a new one with `[dateFormat, timeFormat]` as a dependency array) will simplify the code and make it more readable.

Reply:
Hi @M-i-k-e-l,

Thanks for reviewing. Definitely makes sense to have it in the next major release.
I've made the changes to the code as you've suggested. Let me know if there's anything else you'd like me to change.

Reply:
Looks great!